barretenberg/cpp/src/barretenberg/vm2/simulation/concrete_dbs.hpp

```diff
@@ -80,6 +82,9 @@ class MerkleDB final : public HighLevelMerkleDBInterface {
     LowLevelMerkleDBInterface& as_unconstrained() const override { return raw_merkle_db; }
 
   private:
+    bool nullifier_exists_internal(std::optional<AztecAddress> contract_address, const FF& nullifier) const;
```

Comment:
If C++ lets you, this could be probably be a `std::optional<const AztecAddress&> contract_address`.